### ISSUE TITLE
FluentMark causing Eclipse to crash

### ISSUE THREAD

**Issue by jouvin** [grosenberg/fluentmark#7]
I have been using FluentMark successfully in the last months but starting yesterday, editing a file (probably with an incorrect MD content), I started to get NullPointer exceptions and trying to restart Eclipse, I found it crashing everytime. The only workaround I found was to uninstal FluentMark from another workspace.

The workspace log is:
```
!ENTRY org.eclipse.text 4 2 2017-03-22 22:04:14.414
!MESSAGE Problems occurred when invoking code from plug-in: "org.eclipse.text".
!STACK 0
java.lang.NullPointerException
	at net.certiv.fluentmark.model.PageRoot.addToParent(PageRoot.java:451)
	at net.certiv.fluentmark.model.PageRoot.parse(PageRoot.java:359)
	at net.certiv.fluentmark.model.PageRoot.updateModel(PageRoot.java:139)
	at net.certiv.fluentmark.model.PageRoot.documentChanged(PageRoot.java:86)
	at net.certiv.fluentmark.editor.FluentMkEditor.fireDocumentChanged(FluentMkEditor.java:962)
	at net.certiv.fluentmark.editor.FluentMkEditor$1.documentChanged(FluentMkEditor.java:186)
	at org.eclipse.jface.text.AbstractDocument.doFireDocumentChanged2(AbstractDocument.java:743)
	at org.eclipse.jface.text.AbstractDocument.doFireDocumentChanged(AbstractDocument.java:712)
	at org.eclipse.jface.text.AbstractDocument.doFireDocumentChanged(AbstractDocument.java:696)
	at org.eclipse.jface.text.AbstractDocument.fireDocumentChanged(AbstractDocument.java:770)
	at org.eclipse.jface.text.AbstractDocument.replace(AbstractDocument.java:1101)
	at org.eclipse.core.internal.filebuffers.SynchronizableDocument.replace(SynchronizableDocument.java:173)
	at org.eclipse.jface.text.AbstractDocument.replace(AbstractDocument.java:1119)
	at org.eclipse.core.internal.filebuffers.SynchronizableDocument.replace(SynchronizableDocument.java:161)
	at org.eclipse.jface.text.DefaultDocumentAdapter.replaceTextRange(DefaultDocumentAdapter.java:233)
	at org.eclipse.swt.custom.StyledText.modifyContent(StyledText.java:7374)
	at org.eclipse.swt.custom.StyledText.sendKeyEvent(StyledText.java:8225)
	at org.eclipse.swt.custom.StyledText.doContent(StyledText.java:2530)
	at org.eclipse.swt.custom.StyledText.handleKey(StyledText.java:6018)
	at org.eclipse.swt.custom.StyledText.handleKeyDown(StyledText.java:6049)
	at org.eclipse.swt.custom.StyledText$7.handleEvent(StyledText.java:5733)
	at org.eclipse.swt.widgets.EventTable.sendEvent(EventTable.java:84)
	at org.eclipse.swt.widgets.Display.sendEvent(Display.java:4418)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1079)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1103)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1088)
	at org.eclipse.swt.widgets.Widget.sendKeyEvent(Widget.java:1130)
	at org.eclipse.swt.widgets.Widget.sendKeyEvent(Widget.java:1126)
	at org.eclipse.swt.widgets.Widget.wmChar(Widget.java:1547)
	at org.eclipse.swt.widgets.Control.WM_CHAR(Control.java:4918)
	at org.eclipse.swt.widgets.Canvas.WM_CHAR(Canvas.java:364)
	at org.eclipse.swt.widgets.Control.windowProc(Control.java:4799)
	at org.eclipse.swt.widgets.Canvas.windowProc(Canvas.java:359)
	at org.eclipse.swt.widgets.Display.windowProc(Display.java:5110)
	at org.eclipse.swt.internal.win32.OS.DispatchMessageW(Native Method)
	at org.eclipse.swt.internal.win32.OS.DispatchMessage(OS.java:2552)
	at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:3822)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$4.run(PartRenderingEngine.java:1121)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:336)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.run(PartRenderingEngine.java:1022)
	at org.eclipse.e4.ui.internal.workbench.E4Workbench.createAndRunUI(E4Workbench.java:150)
	at org.eclipse.ui.internal.Workbench$5.run(Workbench.java:693)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:336)
	at org.eclipse.ui.internal.Workbench.createAndRunWorkbench(Workbench.java:610)
	at org.eclipse.ui.PlatformUI.createAndRunWorkbench(PlatformUI.java:148)
	at org.eclipse.ui.internal.ide.application.IDEApplication.start(IDEApplication.java:138)
	at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:196)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:134)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:104)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:388)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:243)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at org.eclipse.equinox.launcher.Main.invokeFramework(Main.java:673)
	at org.eclipse.equinox.launcher.Main.basicRun(Main.java:610)
	at org.eclipse.equinox.launcher.Main.run(Main.java:1519)
```
I attach the file causing FluentMark to crash: [coding_style.md](https://github.com/grosenberg/fluentmark/files/865052/coding_style.md.txt)

**Comment by grosenberg** [comment]
Fixed pending next release. Workaround is to not use underlined (old-style) headings.

**Comment by jouvin** [comment]
Thanks! That's easy, I don't use them normally but in this case I imported a page converted with `pandoc` and `pandoc` converted the original `#`-based headings to `_` headings (conveting from GH markdown to markdown, to get the proper spacing between paragraphs...).

**Comment by grosenberg** [comment]
Fixed with v0.10.0.
